app/src/main/java/com/keylesspalace/tusky/fragment/TimelineFragment.kt

```diff
@@ -0,0 +1,1273 @@
+import com.uber.autodispose.android.lifecycle.AndroidLifecycleScopeProvider.from
+import com.uber.autodispose.autoDispose
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.android.schedulers.AndroidSchedulers
+import retrofit2.Response
+import java.io.IOException
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+
+class TimelineFragment : SFragment(), OnRefreshListener, StatusActionListener, Injectable, ReselectableFragment, RefreshableFragment {
+
+    @Inject
+    lateinit var eventHub: EventHub
+
+    @Inject
+    lateinit var timelineRepo: TimelineRepository
+
+    @Inject
+    lateinit var accountManager: AccountManager
+
+    private val binding by viewBinding(FragmentTimelineBinding::bind)
+
+    private var kind: Kind? = null
+    private var id: String? = null
+    private var tags: List<String> = emptyList()
+
+    private lateinit var adapter: TimelineAdapter
+
+    private var isSwipeToRefreshEnabled = true
+    private var isNeedRefresh = false
+
+    private var eventRegistered = false
+
+    /**
+     * For some timeline kinds we must use LINK headers and not just status ids.
+     */
+    private var nextId: String? = null
+    private var layoutManager: LinearLayoutManager? = null
+    private var scrollListener: EndlessOnScrollListener? = null
+    private var filterRemoveReplies = false
+    private var filterRemoveReblogs = false
+    private var hideFab = false
+    private var bottomLoading = false
+    private var didLoadEverythingBottom = false
+    private var alwaysShowSensitiveMedia = false
+    private var alwaysOpenSpoiler = false
+    private var initialUpdateFailed = false
+
+    private val statuses = PairedList<Either<Placeholder, Status>, StatusViewData> { input ->
+        val status = input.asRightOrNull()
+        if (status != null) {
+            ViewDataUtils.statusToViewData(
+                    status,
+                    alwaysShowSensitiveMedia,
+                    alwaysOpenSpoiler
+            )
+        } else {
+            val (id1) = input.asLeft()
+            StatusViewData.Placeholder(id1, false)
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        val arguments = requireArguments()
+        kind = Kind.valueOf(arguments.getString(KIND_ARG)!!)
+        if (kind == Kind.USER || kind == Kind.USER_PINNED || kind == Kind.USER_WITH_REPLIES || kind == Kind.LIST) {
+            id = arguments.getString(ID_ARG)!!
+        }
+        if (kind == Kind.TAG) {
+            tags = arguments.getStringArrayList(HASHTAGS_ARG)!!
+        }
+
+        isSwipeToRefreshEnabled = arguments.getBoolean(ARG_ENABLE_SWIPE_TO_REFRESH, true)
+
+        val preferences = PreferenceManager.getDefaultSharedPreferences(activity)
+        val statusDisplayOptions = StatusDisplayOptions(
+                preferences.getBoolean("animateGifAvatars", false),
```

Comment:
should we use `PrefKeys` as long as we are here?

Reply:
yes, good point